### PR TITLE
Code refactoring along with new design updates

### LINKED ISSUE #165: Weird alignment in error log
screen shot 2018-09-18 at 10 14 55

### PR DESCRIPTION
**Pull request by ydlamba**
The aim is to finalize the design, use consistent styling, split larger components, improve UX which will fix #165, #164, #156, #147. 

This is currently **WIP**. I'll add the screenshots here once I'm done.

### CODE REVIEW

**Comment by domoritz** [comment]
Sweet. Please also ping me when this is ready for a review.

**Comment by ydlamba** [comment]
Rebased. Resuming work on this.

**Comment by domoritz** [comment]
Sweet. Looking forward to seeing a screenshot soon.

File: src/components/header/renderer.tsx [hunk 1]
```diff
@@ -616,109 +627,20 @@ class Header extends React.Component<Props, State> {
...
-                  </div>
-                </div>
-              ),
-            ]}
-          </PortalWithState>
-
+          <Modal
+            button={examplesButton}
+            header={buttonGroup}
+            content={this.state.showVega ? vega('closePortal') : vegalite('closePortal')}
```
Comment (domoritz) [inline src/components/header/renderer.tsx]: We need to pass the actual function in here

**Pull request closed without merging**